### PR TITLE
[2.7.2] Update PSS docs

### PR DESCRIPTION
**Pull request by macedogm**
This PR proposes some updates to https://github.com/rancher/rancher-docs/pull/423, including:

1. Updated PSS policies descriptions based on https://github.com/rancher/rancher/issues/40761.
2. Updated list of exempt namespaces and example policy based on Rancher provided `rancher-restricted`. See https://github.com/rancher/rancher/blob/release/v2.7/pkg/data/management/podadmissionconfigurationtemplate_data.go#L52-L67.

### CODE REVIEW

**Review by jiaqiluo — APPROVED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (cbron) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: Can you clarify why all of these were removed ? Would like @crobby to confirm changes as well before we merge.

Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: Removed because they are not in the list of ns currently exempt in the policy that we provide in Rancher - https://github.com/rancher/rancher/blob/release/v2.7/pkg/data/management/podadmissionconfigurationtemplate_data.go#L12-L33.

CC @crobby

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md
Comment (btat) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md:9]:
```suggestion
[Pod Security Policies (PSPs)](https://kubernetes.io/docs/concepts/security/pod-security-policy/) are objects that control security-sensitive aspects of pod specification (like root privileges).
```

**Review by btat — APPROVED** [review]
LGTM. Just a small styling consistency suggestion.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (pjbgf) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:60]:
```suggestion
The listed namespaces are already exempt in the Rancher provided `rancher-restricted` PSS policy. A sample Admission Configuration with all the required exemptions to run Rancher can be found [here](psa-restricted-exemptions.yaml).
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md
Comment (pjbgf) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md:7]: We may want to highlight that both co-existed in 1.24 (by default), even though it was [Beta](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) back then.

```suggestion
Pod Security Policy is only available in Kubernetes until v1.24. [Pod Security Standards](pod-security-standards.md) is the built-in alternative, available from Kubernetes v1.24.
:::
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (cbron) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: Ahh, that is because there are two lists. The template is for all downstream clusters and includes things like feature charts namespaces. There is a second list around system namespaces which is meant for local cluster: https://github.com/rancher/rancher/blob/release/v2.7/pkg/settings/setting.go#L27
Something like "cattle-impersonation-system" would only ever be on local.

Comment (cbron) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: Though @crobby can you remove `rancher-operator-system`, the operator has been removed completely in 2.7.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:9]:
```suggestion
Make sure that you migrate all Pod Security Policies. This should grant equivalent security controls through [Pod Security Admissions](https://kubernetes.io/docs/concepts/security/pod-security-admission/), or through solutions such as [Gatekeeper](https://github.com/open-policy-agent/gatekeeper), [Kubewarden](https://www.kubewarden.io/), [Kyverno](https://kyverno.io/), and [NeuVector](https://neuvector.com/). Remove all PodSecurityPolicy objects from the cluster.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:12]:
```suggestion
You must add your new policy enforcement mechanisms _before_ you remove the PodSecurityPolicy objects. If you don't, you may create an opportunity for privilege escalation attacks within the cluster.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:17]: In some places internally we use PSACT, but technically it's `kind: PodSecurityAdmissionConfiguration` when users go to create one. Imho I'd prefer something like _Pod Security Admission (PSA) configuration template_ or _PSA configuration template_ on first mention, then _template_ for short.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:18]:
```suggestion
- `rancher-privileged`: This is the most permissive configuration and doesn't restrict the behavior of the pod, allowing for known privilege escalations. This policy has no exemptions.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:19]:
```suggestion
- `rancher-restricted`: This configuration is heavily restricted and follows current best practices for hardening pods. There are namespace-level exemptions for Rancher components, as described below.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:36]: There's some slippage here between PSA and PSS, as the heading uses "PSA-restricted" but the body text talks of PSS policies. They're closely related concepts with similar acronyms, and a reader might get confused on which is which. Maybe we can avoid the acronyms here by using "security policy"? Example: "[...] a restrictive security policy by default, such as the `rancher-restricted` policy".

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:60]:
```suggestion
The listed namespaces are already exempt in the built-in Rancher `rancher-restricted` security policy. Refer to this [sample Admission Configuration](psa-restricted-exemptions.yaml), which has all the exemptions you need to run Rancher.
```

**Review by martyav — CHANGES_REQUESTED** [review]
I made some style suggestions and left some questions about terminology (PSAC or PSACT or PSA configuration template? PSS policy or security policy?)

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (crobby) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: Yes, that all makes sense to me.

Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: @crobby and @cbron if you are fine with the updates, can you please approve the PR?

**Comment by macedogm** [comment]
> I made some style suggestions and left some questions about terminology (PSAC or PSACT or PSA configuration template? PSS policy or security policy?)

@martyav thank you for the excellent comments and suggestions. I simplified the terminology as you suggested:
- PSAC or PSACT or PSA configuration template -> `PSA configuration template` then simply `template`
- PSS or PSS policy -> `security policy`

Hopefully I address all the remaining open points now. PTAL.

**Review by martyav — APPROVED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:64]: @cbron and @crobby I updated the namespace exemption list, based on our offline discussion, through a separate commit to make it easier for review. PTAL.

**Comment by macedogm** [comment]
@martyav I re-requested your review, because we had to update the namespace exemption list.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:74]:
```suggestion
Rancher, some Rancher owned charts, and RKE2 and K3s distributions all use these namespaces. A subset of the listed namespaces are already exempt in the built-in Rancher `rancher-restricted` policy, for use in downstream clusters. For a complete template which has all the exemptions you need to run Rancher, please refer to this [sample Admission Configuration](psa-restricted-exemptions.yaml).
```

**Review by martyav — CHANGES_REQUESTED** [review]
Some minor changes requested on second look

**Review by crobby — APPROVED** [review]
lgtm

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md:6]:
```suggestion
Pod Security Policy is only available in Kubernetes until v1.24. [Pod Security Standards](pod-security-standards.md) are the built-in alternative.
```

**Comment by btat** [comment]
We may want to consider having the acronym in the headings (although it's not the best in terms of style) as well since headings are what our search looks at.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:5]:
```suggestion
[Pod Security Standards (PSS)](https://kubernetes.io/docs/concepts/security/pod-security-standards/) and [Pod Security admissions (PSA)](https://kubernetes.io/docs/concepts/security/pod-security-admission/) define security restrictions for a broad set of workloads. They replace [Pod Security Policies](https://kubernetes.io/docs/concepts/security/pod-security-policy/) in Kubernetes v1.25 and above.

PSS define security levels for workloads.  PSAs describe requirements for pod security contexts and related fields. PSAs reference PSS levels to define security restrictions.
```

I had a look at https://github.com/rancher/rancher-docs/issues/462, started writing, and realized that the issue overlaps with content in this PR. On this page, we move quickly from PSS to discussing PSA config templates, and on reflection, we need a little more of an explanation of how PSS and PSA relate, so that the section on templates feels more connected.

**Review by martyav — CHANGES_REQUESTED** [review]

**Review by pjbgf — APPROVED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (cbron) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:70]: We don't need this anymore: 
```suggestion
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-restricted-exemptions.yaml
Comment (cbron) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-restricted-exemptions.yaml:50]: Same
```suggestion
```

**Review by cbron — COMMENTED** [review]
Minor things I will try to commit, then lgtm

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:5]: In addition -- this screenshot of the GUI: 
image

The warning in the screenshot states that PSA is the replacement for PSPs, and doesn't mention PSS. Maybe this page should be retitled to Pod Security Admissions & Pod Security Standards, so that it's in sync with the GUI?

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/create-pod-security-policies.md:2]:
```suggestion
title: Pod Security Standards (PSS) & Pod Security Admissions (PSA)
```

See my comment further down about the Rancher GUI.

Also:  It's just hard to talk about either resource without mentioning the other.

It's tempting to pull PSACT content into its own page, since how-to create and edit PSACTs feels like its own topic. That leaves the PSS page with the intro, a section on migration, and a section on Rancher namespace exemptions. Thoughts?

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]:
```suggestion
title: Pod Security Standards (PSS) & Pod Security Admissions (PSA)
```

See my comment further down about the Rancher GUI.

Also:  It's just hard to talk about either resource without mentioning the other.

It's tempting to pull PSACT content into its own page, since how-to create and edit PSACTs feels like its own topic. That leaves the PSS page with the intro, a section on migration, and a section on Rancher namespace exemptions. Thoughts?

Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]: > It's tempting to pull PSACT content into its own page, since how-to create and edit PSACTs feels like its own topic. That leaves the PSS page with the intro, a section on migration, and a section on Rancher namespace exemptions. Thoughts?

@martyav I agree, but I'm not sure what are the plans about expanding the docs further.

@jiaqiluo do you plan to address ^ in https://github.com/rancher/rancher-docs/issues/462?

Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]: I think that a high-level section about PSS/PSA and then a subpage about PSACT would make sense.

Comment (jiaqiluo) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]: hi @macedogm, I agree with your idea.   
@martyav will handle #462 , so I will leave it to him to finalize the structure. 

I think we do not need to hold this PR because of that. We can get this PR merged and use #462  to do the refactor or enhancement.

**Review by jiaqiluo — APPROVED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md
Comment (macedogm) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]: Just a note, before getting Marty's feedback about merging this PR or not, we still need to merge his suggestion above. I haven't done this yet, because it will close this thread (that is still going on).

Comment (martyav) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/pod-security-standards.md:2]: I think it's a good idea to merge this PR if there is nothing else to add, then I can create a working branch off of the 2.7.2 release branch, and open a PR for the PSACT how-to. It'll be easier to review the how-to that way.

**Review by martyav — APPROVED** [review]

**Pull request merged**
